engine/step/step.go

```diff
@@ -493,13 +501,30 @@ func AfterRun(st *Step, values *values.Values, ss StateSetter) {
 				break
 			}
 		}
-		for step, state := range sc.Then {
-			if step == stepRefThis {
-				step = st.Name
+		switch sc.Type {
+		case condition.LOOP:
+			if st.MaxRetries < 0 || st.TryCount <= st.MaxRetries {
+				state := fmt.Sprintf("%s%d", StateLooping, st.TryCount)
```

Comment:
I wonder why we need to have the `TryCount` as the state name here.
Why don't go directly with `LOOPING`, and use the `TryCount` to know how many iterations as been performed ?